pump/server.go

```diff
@@ -72,6 +72,8 @@ type Server struct {
 	gs         *grpc.Server
 	ctx        context.Context
 	cancel     context.CancelFunc
+	pullCtx    context.Context
+	pullCancel context.CancelFunc
```

Comment:
add somethink like `stopPull chan struct{}` and close it to make PullBinlogs quit